packages/orchestration/src/examples/staking-combinations.contract.js

```diff
@@ -0,0 +1,138 @@
+import { VowShape } from '@agoric/vow';
+import { M } from '@endo/patterns';
+import { prepareCombineInvitationMakers } from '../exos/combine-invitation-makers.js';
+import { CosmosOrchestrationInvitationMakersInterface } from '../exos/cosmos-orchestration-account.js';
+import { withOrchestration } from '../utils/start-helper.js';
+import * as flows from './staking-combinations.flows.js';
+
+/**
+ * @import {GuestInterface} from '@agoric/async-flow';
+ * @import {Delegation} from '@agoric/cosmic-proto/cosmos/staking/v1beta1/staking.js';
+ * @import {ContinuingOfferResult} from '@agoric/smart-wallet/src/types.js';
+ * @import {TimerService} from '@agoric/time';
+ * @import {LocalChain} from '@agoric/vats/src/localchain.js';
+ * @import {NameHub} from '@agoric/vats';
+ * @import {Vow} from '@agoric/vow';
+ * @import {Remote} from '@agoric/internal';
+ * @import {Zone} from '@agoric/zone';
+ * @import {CosmosInterchainService} from '../exos/cosmos-interchain-service.js';
+ * @import {OrchestrationTools} from '../utils/start-helper.js';
+ * @import {CosmosOrchestrationAccount} from '../exos/cosmos-orchestration-account.js';
+ */
+
+const emptyOfferShape = harden({
+  // Nothing to give; the funds are deposited offline
+  give: {},
+  want: {}, // UNTIL https://github.com/Agoric/agoric-sdk/issues/2230
+  exit: M.any(),
+});
+
+/**
+ * Orchestration contract to be wrapped by withOrchestration for Zoe.
+ *
+ * @param {ZCF} zcf
+ * @param {{
+ *   agoricNames: Remote<NameHub>;
+ *   localchain: Remote<LocalChain>;
+ *   orchestrationService: Remote<CosmosInterchainService>;
+ *   storageNode: Remote<StorageNode>;
+ *   marshaller: Marshaller;
+ *   timerService: Remote<TimerService>;
+ * }} privateArgs
+ * @param {Zone} zone
+ * @param {OrchestrationTools} tools
+ */
+const contract = async (
+  zcf,
+  privateArgs,
+  zone,
+  { orchestrateAll, vowTools },
+) => {
+  const ExtraInvitationMakerInterface = M.interface('', {
+    DepositAndDelegate: M.call(M.array()).returns(VowShape),
+    UndelegateAndTransfer: M.call(M.array()).returns(VowShape),
+  });
+  /** @type {any} XXX async membrane */
+  const makeExtraInvitationMaker = zone.exoClass(
+    'ContinuingInvitationExampleInvitationMakers',
+    ExtraInvitationMakerInterface,
+    /** @param {GuestInterface<CosmosOrchestrationAccount>} account */
+    account => {
+      return { account };
+    },
+    {
+      DepositAndDelegate() {
+        const { account } = this.state;
+
+        const invP = zcf.makeInvitation(
+          (seat, validatorAddr, amountArg) =>
+            // eslint-disable-next-line no-use-before-define -- defined by orchestrateAll, necessarily after this
+            orchFns.depositAndDelegate(account, seat, validatorAddr, amountArg),
+          'Deposit and delegate',
+          undefined,
+          {
+            give: {
+              Stake: AmountShape,
+            },
+          },
+        );
+
+        return vowTools.watch(invP);
```

Comment:
Thinking on this more, I don't think we need `vowTools.watch` here. We can expect the promise for an Invitation to resolve promptly - it's just the offer handler that needs to return a vow.

If we feel we should return a vow here, we should also make this change in the orchAccount exos. Currently, we're always returning a promise.

The only instance coming to mind where it'd be helpful to return a vow is if a developer wanted to call an `invitationMaker` from _inside_ a flow - it'd need to be a vow to cross the membrane. I'm not sure how common this will be - we don't have any examples that need this currently.

Reply:
You're right it's not necessary. https://github.com/Agoric/agoric-sdk/pull/10044

> We can expect the promise for an Invitation to resolve promptly

I believe "promptly" means in "in the same run". `zcf.makeInvitation` calls out to another vat so I don't think it can be in the same run.

https://github.com/Agoric/agoric-sdk/blob/daff9eba00e9a8299d8b5c50db845c96ebf070d0/packages/zoe/src/contractFacet/zcfZygote.js#L314-L316